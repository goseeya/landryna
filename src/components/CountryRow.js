const CountryRow = ({ countryData }) => {
  return (
    <tr className="country-row">
      <td>{countryData.name}</td>
      <td>{countryData.region}</td>
      <td>
        {!!countryData.area ? Math.floor(countryData.area * 0.3861021586) : 'no data'}
      </td>
      <td>{Math.round((countryData.population / 1000000) * 10) / 10}</td>
    </tr>
  );
};

export default CountryRow;
