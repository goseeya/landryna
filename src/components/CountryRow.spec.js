import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import CountryRow from "./CountryRow";
import * as mockedCountryRow from "../mock/countryRow.json";
import * as mockedCountryRowNullable from "../mock/countryRowNullable.json";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { render } from "@testing-library/react";

configure({ adapter: new Adapter() });

const createFromProps = (props) => {
  <CountryRow countryData={props} />;
};

describe("CountryRow", () => {
  describe("Snapshot", () => {
    it("should match base snapshot", () => {
      const component = create(createFromProps(mockedCountryRow));
      expect(component.toJSON()).toMatchSnapshot();
    });
    it("should render country row", () => {
      const component = shallow(<CountryRow countryData={mockedCountryRow} />);
      expect(component.find(".country-row")).toHaveLength(1);
    });
    it('should display "no data" when area is  null', () => {
      const component = render(
        <table>
          <tbody>
            <CountryRow countryData={mockedCountryRowNullable} />
          </tbody>
        </table>
      );
      expect(component.getByText(/no data/i)).toBeInTheDocument();
    });
  });
});
