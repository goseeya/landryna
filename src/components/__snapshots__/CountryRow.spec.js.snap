// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryRow Snapshot should match base snapshot 1`] = `null`;
